Makefile

```diff
@@ -1,5 +1,52 @@
+all: test check-coverage lint tidy ## Tests, lints and checks coverage
+
+clean:  ## Remove generated files
+
+.PHONY: all clean
+
+# -- Lint ----------------------------------------------------------------------
+lint: ## Lint Go Source Code
+	golangci-lint run
+
+tidy:
+	go mod tidy
+
+.PHONY: lint tidy clean
+
+# -- Test ----------------------------------------------------------------------
 COVERFILE=coverage.out
+COVERAGE = 50
+
+test:
+	go test -coverprofile=$(COVERFILE) -tags codeanalysis ./...
+
+check-coverage: test  ## Check that test coverage meets the required level
+	@go tool cover -func=$(COVERFILE) | $(CHECK_COVERAGE) || $(FAIL_COVERAGE)
+
+coverage: test  ## Show test coverage in your browser
+	go tool cover -html=$(COVERFILE)
+
+clean: 
+	rm -f $(COVERFILE)
+
+CHECK_COVERAGE = awk -F '[ \t%]+' '/^total:/ && $$3 < $(COVERAGE) {exit 1}'
+FAIL_COVERAGE = { echo '$(COLOUR_RED)FAIL - Coverage below $(COVERAGE)%$(COLOUR_NORMAL)'; exit 1; }
 
+.PHONY: check-coverage coverage test
```

Comment:
`test` target missing.

Reply:
Sorry, a little bit confused by this. The target for `test` is above I think
```
test: ## Run all tests
	go test -coverprofile=$(COVERFILE) -tags codeanalysis ./...
```

Reply:
yeah, so `.PHONY` needs to hold _all_ targets that don't match any actual file output generation. (this is an artefact from creating/ compiling c executables). Maybe we should talk about this at some point in more detail.

Anyway, long story short, you need to list `test` under `.PHONY` too.

Reply:
Awesome, thanks for the explanation, `test` has been included undernder `.PHONY` in https://github.com/anz-bank/sysl-go/pull/23/commits/a459be5ee642cd9884c84430384c4e82f9e0abdd